src/sentry_core.c

```diff
@@ -85,6 +94,7 @@ sentry_init(sentry_options_t *options)
         SENTRY_WARN("failed to create database directory or there is no write "
                     "access to this directory");
         sentry_options_free(options);
```

Comment:
maybe `goto fail`?

Reply:
ACK, good option here, crosses no important code doing this.